### ISSUE TITLE
Integrity of `skos:preflabel` and `skos:prefdefinition`

### ISSUE THREAD

**Issue by tomhgmns** [solid/data-interoperability-panel#276]
We are currently implementing the interop spec into use.id, but are concerned about relying on `skos:preflabel` and `skos:prefdefinition` of the Access Need Groups to render the UI of the authz agent.

This is because a malicious party might create a mismatch between those two fields and the actual access requests. 

For example, I could present the following access needs group:
- `skos:preflabel`: "Read access to your shopping history"
- In reality, my app asks permission to read the user's medical data

Has the panel considered this situation?

We are thinking to solve this issue by putting a human readable name at the shape tree itself...

**Comment by justinwb** [comment]
Yes this scenario was considered and has been solved already in the shape tree specification. The shape tree definition includes a similar description of what the data is. So regardless of what the access need description is, the data that is being requested is always present it based on the shapetree definition.

**Comment by tomhgmns** [comment]
Thanks for the answer, Justin! We'll rely on the ShapeTree definition then!

**Comment by elf-pavlik** [comment]
Great timing @tomhgmns I'm also implementing it and today should have all the code ready, that combines human-readable information from access need descriptions and shape tree descriptions. I plan to push that code down to sai-js as soon as it fits the need of the authorization agent.

**Comment by tomhgmns** [comment]
Great!

By the way, I just had a call with Laurens about this and during this call it became clear that I forgot to tell about a very important part of our implementation...

By default, we have created several containers where apps can access data. The user has no access to those containers (not even read access). The user only has access to the root.

We've did this to avoid token replay (i.e. apps pretending they are the user).

**Comment by elf-pavlik** [comment]
I'll look into adding what we have discussed here into the Security Considerations section.
